DEV-15145 - Update references of Elasticsearch to Opensearch (#24)
Datatamer/terraform-aws-opensearch: README.md

> # Tamr AWS Elasticsearch Terraform Module

Reviewer: ```suggestion
# Tamr AWS OpenSearch Terraform Module
```

This terraform module creates an OpenSearch domain on AWS.

**Prerequisite**

This module requires an [IAM service linked role](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/slr.html) for OpenSearch on the AWS account.

To create an OpenSearch service role in terraform:
```
resource "aws_iam_service_linked_role" "opensearch-service-role" {
  aws_service_name = "opensearchservice.amazonaws.com"
}
```

OpenSearch Service automatically creates a new OpenSearch service-linked role the first time you create an OpenSearch domain if you have permissions for the iam:CreateServiceLinkedRole action and the legacy Elasticsearch role doesn't exist in your account.

There can be only one service linked role for per AWS account.

> You may run into an error like this when trying to remove the ES service linked role if there is still an OpenSearch domain in the account:

Reviewer: ```suggestion
You may run into an error like this when trying to remove the service linked role if there is still an OpenSearch domain in the account:
```

| ebs\_volume\_size | The size of EBS volumes attached to data nodes (in GB) | `number` | `100` | no |
| ebs\_volume\_type | The type of EBS volumes attached to data nodes | `string` | `"gp2"` | no |
| enable\_http | If set to true, enables SSH | `bool` | `true` | no |
| enable\_https | If set to true, enables SSH | `bool` | `true` | no |
| enforce\_https | Whether or not to require HTTPS on the domain endpoint | `bool` | `true` | no |
| es\_tags | [DEPRECATED: Use `tags` instead] Additional tags to be attached to the OpenSearch domain and associated resources. | `map(string)` | `{}` | no |
| es\_version | Version of OpenSearch to deploy | `string` | `"6.8"` | no |
| ingress\_cidr\_blocks | CIDR blocks to attach to security groups for ingress | `list(string)` | `[]` | no |
| ingress\_security\_groups | Existing security groups to attch to new security groups for ingress | `list(string)` | `[]` | no |
| instance\_count | Number of instances to launch in the OpenSearch domain | `number` | `2` | no |
| instance\_type | Instance type of data nodes in the domain | `string` | `"c5.large.elasticsearch"` | no |

> | kms\_key\_id | The KMS key id to encrypt the Elasticsearch domain with.<br>  If not specified then it defaults to using the aws/es service KMS key | `string` | `null` | no |

Reviewer: Couldn't suggest the edit in the variables.tf file for some reason
```suggestion
| kms\_key\_id | The KMS key id to encrypt the OpenSearch domain with.<br>  If not specified then it defaults to using the aws/OpenSearch service KMS key | `string` | `null` | no |
```

> | log\_group\_name | The name of an existent CloudWatch Log Group that ElasticSearch will publish logs to | `string` | `""` | no |

Reviewer: ```suggestion
| log\_group\_name | The name of an existent CloudWatch Log Group that OpenSearch will publish logs to | `string` | `""` | no |
```
